lib/main.coffee

```diff
@@ -37,48 +28,87 @@ module.exports =
 
   configSub: null
   editorSub: null
-  watching: Object.create null
+  buffers: new Map
   debug: false
 
   activate: (state) ->
     return if atom.inSpecMode() or not atom.inDevMode()
 
     @configSub = atom.config.observe 'grammar-live-reload.enabled', (enabled) =>
-      return @editorSub?.dispose() unless enabled
 
-      reload = @reload.bind this
+      unless enabled
+        @editorSub?.dispose()
+        @editorSub = null
+        @buffers.forEach (subs) -> subs.dispose()
+        @buffers.clear()
+        return
+
+      resolveSymlink = (filePath) ->
```

Comment:
please extract the functions

Reply:
It's not used outside that scope. Isn't it better to keep it close to where it is used, since minor perf improvements to config changes are not a necessity?

Reply:
please extract the functions.